daft/runners/pyrunner.py

```diff
@@ -107,27 +108,118 @@ def wait(self) -> None:
         pass
 
 
-class PyActorPool:
-    initialized_stateful_udfs_process_singleton: dict[str, UserProvidedPythonFunction] | None = None
+@dataclass
+class PyRunnerResources:
+    num_cpus: float
+    gpus: dict[str, float]
```

Comment:
Thoughts on potentially just limiting this to integer values?

Reply:
What would be the rationale for that? We currently support partial GPU requests so it would be kind of a breaking change.